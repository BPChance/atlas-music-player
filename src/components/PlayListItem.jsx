export default function PlayListItem() {
  return (
    <div>
      <p>song title</p>
      <p>artist</p>
      <p>length</p>
    </div>
  );
}
