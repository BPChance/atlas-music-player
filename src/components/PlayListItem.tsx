type PlayListItemProps = {
  title: string;
  artist: string;
  duration: number;
  selected: boolean;
};

function formatDuration(seconds: number): string {
  const mins = Math.floor(seconds / 60);
  const secs = seconds % 60;
  return `${mins}:${secs < 10 ? "0" : ""}${secs}`;
}

export default function PlayListItem({
  title,
  artist,
  duration,
  selected,
}: PlayListItemProps) {
  return (
    <div
      className={`flex w-full items-center justify-between rounded-lg px-4 py-1 ${
        selected ? "bg-supablue dark:bg-supapurp text-supapink shadow-2xl" : ""
      }`}
    >
      <div className="flex max-w-[75%] flex-col">
        <p className="font-semibold">{title}</p>
        <p className="text-sm text-[#94A3B8]">{artist}</p>
      </div>
      <p className="shrink-0 pl-4 text-sm whitespace-nowrap text-[#94A3B8]">
        {formatDuration(duration)}
      </p>
    </div>
  );
}
