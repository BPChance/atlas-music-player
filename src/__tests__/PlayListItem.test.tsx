import { render } from "@testing-library/react";
import PlayListItem from "../components/PlayListItem";
import { describe, expect, it } from "vitest";

describe("PlayListItem", () => {
  it("renders correctly", () => {
    const { container } = render(
      <PlayListItem
        title="test song"
        artist="test artist"
        duration={120}
        selected={true}
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it("renders unselected item", () => {
    const { container } = render(
      <PlayListItem
        title="Another Song"
        artist="Another Artist"
        duration={95}
        selected={false}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
