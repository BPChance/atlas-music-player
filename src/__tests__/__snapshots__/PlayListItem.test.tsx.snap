// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PlayListItem > renders correctly 1`] = `
<div>
  <div
    class="flex w-full items-center justify-between rounded-lg px-4 py-1 bg-supablue dark:bg-supapurp text-supapink shadow-2xl"
  >
    <div
      class="flex max-w-[75%] flex-col"
    >
      <p
        class="font-semibold"
      >
        test song
      </p>
      <p
        class="text-sm text-[#94A3B8]"
      >
        test artist
      </p>
    </div>
    <p
      class="shrink-0 pl-4 text-sm whitespace-nowrap text-[#94A3B8]"
    >
      2:00
    </p>
  </div>
</div>
`;

exports[`PlayListItem > renders unselected item 1`] = `
<div>
  <div
    class="flex w-full items-center justify-between rounded-lg px-4 py-1 "
  >
    <div
      class="flex max-w-[75%] flex-col"
    >
      <p
        class="font-semibold"
      >
        Another Song
      </p>
      <p
        class="text-sm text-[#94A3B8]"
      >
        Another Artist
      </p>
    </div>
    <p
      class="shrink-0 pl-4 text-sm whitespace-nowrap text-[#94A3B8]"
    >
      1:35
    </p>
  </div>
</div>
`;
